scalafix-tests/input/src/main/scala/test/ImplicitClassPrivateParams.scala

```diff
@@ -0,0 +1,56 @@
+/*
+rules = ImplicitClassPrivateParams
+ */
+package test
+
+object ImplicitClassPrivateParams {
+  implicit class ValueClassXtensionVal(val int: Int) extends AnyVal {
+    def doubled: Int = int + int
+  }
+
+  implicit class ValueClassXtensionAnnotatedVal(@transient val str: String) extends AnyVal {
+    def doubled: String = str + str
+  }
+
+  implicit class XtensionVal(val str: String) {
+    def doubled: String = str + str
+  }
+
+  implicit class XtensionAnnotatedVal(@transient val str: String) {
+    def doubled: String = str + str
+  }
+
+  implicit class XtensionAnnotatedFinalVal(@transient final val str: String) {
+    def doubled: String = str + str
+  }
+
+  implicit class XtensionVar(var str: String) {
```

Comment:
We can ignore `var` cases since those are less commonly (if ever) used for implicit classes.

Reply:
Isn't it best to keep these as non reg tests ? Unless you think this is being overzealous.

Reply:
Negative tests are good :) I probably don't write enough of them. We don't need too many here however since the implementation guards against `val`